### PR TITLE
Make the browser example watch independent from start

### PR DESCRIPTION
**Pull request by hexa00**
This patch makes it so that npm start no longer start watchers.

So that the following developement workflow is now possible:

#Watch the backend, frontend and start the file-depencency-updater 
npm run watch 

#Run the backend and frontend
npm start

Signed-off-by: Antoine Tremblay <antoine.tremblay@ericsson.com>

### CODE REVIEW

File: examples/browser/package.json [hunk 1]
```diff
@@ -21,11 +21,12 @@
...
     "build:frontend": "webpack && cp src/frontend/index.html lib/frontend",
     "lint": "tslint -c ../../config/tslint/tslint.json --project ./tsconfig.json",
-    "start": "concurrently --names watch-deps,backend,webpack-server --prefix \"[{name}]\" \"npm run watch:dependencies\" \"npm run start:backend\" \"npm run start:frontend\"",
+    "watch": "concurrently --names watch-deps,watch-backend,watch-frontend --prefix \"[{name}]\" \"npm run watch:dependencies\" \"npm run watch:backend\" \"npm run watch:frontend\"",
+    "start": "concurrently --names backend,frontend --prefix \"[{name}]\" \"npm run start:backend\" \"npm run start:frontend\"",
     "start:backend": "node ./lib/backend/main.js",
     "start:frontend": "webpack-dev-server --open",
     "cold:start": "npm run clean && npm run build && npm start",
-    "watch:backend": "tsc --watch",
+    "watch:backend": "tsc --watch -p ../../",
```
Comment (akosyakov) [inline examples/browser/package.json]: Why ` -p ../../`?

Comment (hexa00) [inline examples/browser/package.json]: Yes this is wrong, my intention was to watch theia-core, I'll add a new watch:core with tsc --watch -p ../../ and leave the watch:backend as it was

Comment (akosyakov) [inline examples/browser/package.json]: why not run `watch` on `theia-core` itself?

Comment (hexa00) [inline examples/browser/package.json]: Well it's another shell & another command it's running the file-updater anyway so like this all you would have to do is npm run watch and all would be taken care of ?

Comment (akosyakov) [inline examples/browser/package.json]: ok, make sense

**Comment by hexa00** [comment]
With this however it makes it very apparent that the file-updater is not working properly.

We get into the same kind of errors we had with chokidar like :

```
[watch-frontend] SyntaxError: Unexpected end of JSON input
[watch-frontend]     at JSON.parse (<anonymous>)
[watch-frontend]     at /home/eantotr/ownCloud/work/theia/examples/browser/node_modules/source-map-loader/index.js:44:22
[watch-frontend]     at tryToString (fs.js:449:3)
[watch-frontend]     at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)
[watch-frontend] 
ch-backend] node_modules/theia-core/lib/application/browser/application.d.ts(3,33): error TS2306: File '/home/eantotr/ownCloud/work/theia/examples/browser/node_modules/theia-core/lib/application/common/command.d.ts' is not a module.
[watch-backend] node_modules/theia-core/lib/application/browser/application.d.ts(4,36): error TS2306: File '/home/eantotr/ownCloud/work/theia/examples/browser/node_modules/theia-core/lib/application/common/keybinding.d.ts' is not a module.
[watch-backend] node_modules/theia-core/lib/application/browser/menu/menu-plugin.d.ts(3,33): error TS2306: File '/home/eantotr/ownCloud/work/theia/examples/browser/node_modules/theia-core/lib/application/common/command.d.ts' is not a module.
[watch-backend] node_modules/theia-core/lib/application/browser/menu/menu-plugin.d.ts(4,36): error TS2306: File '/home/eantotr/ownCloud/work/theia/examples/browser/node_modules/theia-core/lib/application/common/keybinding.d.ts' is not a module.

```
This makes it unusable for me :(

I think we should accept this pull request and work on the file-updater separately

**Comment by akosyakov** [comment]
@hexa00 Will you rebase or close it?

**Comment by hexa00** [comment]
Yes I just rebased and will merge but I removed the localdep watch and the core watch from there since it was just not working.

So watch only watches the frontend/backend and the dev is responsible to watch the rest elsewhere or run a build...

Actually running a build I think should be the preferred way, I'm still experimenting before writing something I'm confident about in the dev doc.

**Pull request merged**